libs/libcommon/src/libcommon/exceptions.py

```diff
@@ -496,6 +497,19 @@ def __init__(self, message: str, cause: Optional[BaseException] = None):
         )
 
 
+class SplitParquetSchemaMismatchError(CacheableError):
+    """The split parquet files have different schema."""
```

Comment:
same comment as below: the error message could be more explicit for the end user